lxd/db/node.go

```diff
@@ -1111,6 +1111,85 @@ func whereClause(conditions []inCondition, args *[]any) string {
 	return b.String()
 }
 
+// OverridableConfig is a config key that could be set on either a specific
+// cluster member or for the whole cluster.
+type OverridableConfig struct {
+	ClusterValue sql.NullString
+	MemberValue  sql.NullString
+}
+
+// GetMemberConfigWithGlobalDefault returns a mapping of
+// [clusterMemberName][key]cfg, with filters clusterMemberNames and keys. If
+// either slice is nil/empty, all clusterMembers/keys are returned.
+func (c *ClusterTx) GetMemberConfigWithGlobalDefault(ctx context.Context, clusterMemberNames []string, keys []string) (map[string]map[string]OverridableConfig, error) {
```

Comment:
Is this function just used for access the effective config, or are you using it to display the config to the end user?

If the former, have you considered returning a map of effective config values for each cluster member, rather than requiring the caller to consult the ClusterValue and the MemberValue.

Also we dont use sqli.NullString for config values, unset keys are just not present in the map.